### ISSUE TITLE
jinja2._stringdefs uses much RAM

### ISSUE THREAD

**Issue by methane** [pallets/jinja#666]
`jinja2._stringdefs` makes large strings for each unicode category.
They eat 100KB+ RAM.  But none of them are used.

Only `xid_start` and `xid_continue` are used by `jinja2.lexer`.
Would you remove category strings?

## Your Environment
* Python version: Python 3.6
* Jinja version: 2.9.4

**Comment by ThiefMaster** [comment]
100kb is not really "much ram". However, the module could probably be removed from `sys.modules` after this code has been executed:

```python
try:
    compile('föö', '<unknown>', 'eval')
except SyntaxError:
    name_re = re.compile(r'\b[a-zA-Z_][a-zA-Z0-9_]*\b')
else:
    from jinja2 import _stringdefs
    name_re = re.compile(r'[%s][%s]*' % (_stringdefs.xid_start,
                                         _stringdefs.xid_continue))
```

**Comment by davidism** [comment]
Solved more by #731.
